Forge.TreeWalker/src/TreeWalkerSession.cs

```diff
@@ -321,18 +358,40 @@ public async Task<string> WalkTree(string treeNodeKey)
                     }
                     finally
                     {
-                        // Always call this callback, whether or not visit node was successful.
-                        await this.Parameters.Callbacks.AfterVisitNode(
+                        if (this.Parameters.CallbacksV2 != null)
+                        {
+                            // When CallbacksV2 is not null, continues with the new ITreeWalkerCallbacksV2.
+                            treeNodeContext = new TreeNodeContext(
```

Comment:
>treeNodeContext = new TreeNodeContext( [](start = 28, length = 38)

Let's add a comment here since its not obvious why we are recreating the TreeNodeContext.

// Recreating the TreeNodeContext here to reevaluate TreeNode.Properties.

Reply:
i was thinking the same before iteration #5, but forgot to add it. Fixed.

---
In reply to: [505913873](https://github.com/microsoft/Forge/pull/32#discussion_r505913873) [](ancestors = 505913873)